web/src/main/java/com/epam/rft/atsy/web/controllers/rest/PasswordChangeController.java

```diff
@@ -0,0 +1,110 @@
+package com.epam.rft.atsy.web.controllers.rest;
+
+import com.epam.rft.atsy.service.PasswordChangeService;
+import com.epam.rft.atsy.service.UserService;
+import com.epam.rft.atsy.service.domain.PasswordChangeDTO;
+import com.epam.rft.atsy.service.domain.PasswordHistoryDTO;
+import com.epam.rft.atsy.service.domain.UserDTO;
+import com.epam.rft.atsy.service.exception.passwordchange.PasswordValidationException;
+import com.epam.rft.atsy.service.passwordchange.validation.PasswordValidator;
+import com.epam.rft.atsy.service.security.UserDetailsAdapter;
+import com.epam.rft.atsy.web.CurrentUser;
+import com.epam.rft.atsy.web.exceptionhandling.RestResponse;
+import com.epam.rft.atsy.web.mapper.RuleValidationExceptionMapper;
+import com.epam.rft.atsy.web.messageresolution.MessageKeyResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Date;
+import javax.annotation.Resource;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(value = "/secure/password")
+public class PasswordChangeController {
+
+  public static final String LOGIN_BACKEND_VALIDATION = "login.backend.validation";
+
+  private static Logger
+      logger =
+      LoggerFactory.getLogger(PasswordChangeController.class);
+
+  @Resource
+  MessageKeyResolver messageKeyResolver;
+
+  @Resource
+  PasswordChangeService passwordChangeService;
+
+  @Resource
+  UserService userService;
+
+  @Resource
+  RuleValidationExceptionMapper ruleValidationExceptionMapper;
+
+  @Autowired
+  PasswordValidator passwordValidator;
```

Comment:
Should consistently use Autowired or Resource depending on which is used most in the project.